jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -608,7 +608,7 @@ private static void ensureProperties(boolean isInitialization) {
 	initializedProperties.put("sun.jnu.encoding", platformEncoding); //$NON-NLS-1$
 	initializedProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
 	/*[IF JAVA_SPEC_VERSION >= 17]*/
-	initializedProperties.put("native.encoding", (fileEncoding == null) ? platformEncoding : fileEncoding); //$NON-NLS-1$
+	initializedProperties.put("native.encoding", platformEncoding); //$NON-NLS-1$
```

Comment:
Is it JDK18+ behaviours, JDK17 need stay same?

Reply:
It was just wrong for jdk17 but we didn't notice. According to JEP 400 this is how it works, and testing on jdk17 RI shows the same.

Reply:
I guess I need to move this down so it's not overwritten by a value set on the command line.

Reply:
Updated.